Mathlib/RingTheory/Valuation/Discrete/Basic.lean

```diff
@@ -3,52 +3,67 @@ Copyright (c) 2025 María Inés de Frutos-Fernández, Filippo A. E. Nuccio. All
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: María Inés de Frutos-Fernández, Filippo A. E. Nuccio
 -/
-import Mathlib.Algebra.GroupWithZero.Int
+import Mathlib.Algebra.Order.Group.Cyclic
+import Mathlib.Algebra.Order.Group.Units
 import Mathlib.RingTheory.Valuation.Basic
 
 /-!
 # Discrete Valuations
 
-A valuation `v : A → ℤₘ₀` on a ring `A` is said to be a (normalized) discrete valuation if
-`ofAdd (-1 : ℤ)` belongs to the image of `v`. Note that valuations in Mathlib are multiplicative;
-if `a : A → ℤ ∪ {infty}` is the additive valuation associated to `v`, this is equivalent to asking
-that `1 : ℤ` belongs to the image of `a`.
+Given a linearly ordered commutative group with zero `Γ` such that `Γˣ` is nontrivial cyclic, a
+valuation `v : A → Γ` on a ring `A` is *discrete*, if `genLTOne Γˣ` belongs to the image, where
+`genLTOne Γˣ` is the generator of `Γˣ` that is `< 1`. When `Γ := ℤₘ₀` (defined in
+`Multiplicative.termℤₘ₀`), `genLTOne Γˣ = ofAdd (-1)` and the condition of being discrete is
+equivalent to asking that `ofAdd (-1 : ℤ)` belongs to the image, in turn equivalent to asking that
+`1 : ℤ` belongs to the image of the corresponding *additive* valuation.
+
 
 ## Main Definitions
-* `IsDiscrete`: We define a valuation to be discrete if it is `ℤₘ₀`-valued and `ofAdd (-1 : ℤ)`
-belongs to the image.
+* `IsDiscrete`: We define a `Γ`-valued valuation `v` to be discrete if `Γˣ` is cyclic and
+  `genLTOne Γˣ` belongs to the image of `v`
 
 ## TODO
-* Define (pre)uniformizers for nontrivial `ℤₘ₀`-valued valuations.
+* Define (pre)uniformizers for nontrivial discrete valuations.
 * Relate discrete valuations and discrete valuation rings.
-
 -/
 
 namespace Valuation
 
-open Function Multiplicative Set
+open Function Set LinearOrderedCommGroup
 
-variable {A : Type*} [Ring A]
+variable {Γ : Type*} [LinearOrderedCommGroupWithZero Γ]
 
-/-- A valuation `v` on a ring `A` is (normalized) discrete if it is `ℤₘ₀`-valued and
-  `ofAdd (-1 : ℤ)` belongs to the image. Note that the latter is equivalent to
-  asking that `1 : ℤ` belongs to the image of the corresponding additive valuation. -/
-class IsDiscrete (v : Valuation A ℤₘ₀) : Prop where
-  one_mem_range : (↑(ofAdd (-1 : ℤ)) : ℤₘ₀) ∈ range v
+variable {A : Type*} [Ring A] (v : Valuation A Γ)
+
+/-- Given a linearly ordered commutative group with zero `Γ` such that `Γˣ` is
+nontrivial cyclic, a valuation `v : A → Γ` on a ring `A` is *discrete*, if
+`genLTOne Γˣ` belongs to the image. Note that the latter is equivalent to
+asking that `1 : ℤ` belongs to the image of the corresponding additive valuation. -/
+class IsDiscrete [IsCyclic Γˣ] [Nontrivial Γˣ] : Prop where
```

Comment:
You don't have to ask for `Nontrivial` because the axiom implies it. And you don't have to ask for IsCyclic either, for the same reason! So shouldn't you just not ask for them?
```
class IsDiscrete : Prop where
  exists_generator_lt_one : ∃ (γ : Γˣ), Subgroup.zpowers γ = ⊤ ∧ γ < 1 ∧ ↑γ ∈ range v
```

Reply:
Oh sure, thanks!